packages/ckeditor5-widget/src/widget.js

```diff
@@ -146,12 +146,15 @@ export default class Widget extends Plugin {
 			// But at least triple click inside nested editable causes broken selection in Safari.
 			// For such event, we select the entire nested editable element.
 			// See: https://github.com/ckeditor/ckeditor5/issues/1463.
-			if ( env.isSafari && domEventData.domEvent.detail >= 3 ) {
+			if ( ( env.isSafari || env.isGecko ) && domEventData.domEvent.detail >= 3 ) {
 				const mapper = editor.editing.mapper;
-				const modelElement = mapper.toModelElement( element );
+				const viewElement = element.is( 'attributeElement' ) ?
+					element.findAncestor( element => !element.is( 'attributeElement' ) ) : element;
```

Comment:
What if there's a `widget > nested editable > block-quote > link` structure? Feels like `viewElement` will be the blockquote and the entire nested editable will not be selected. I think `isInsideNestedEditable()` could become `getClosestNestedEditable()` and you could re-use its output here.

Reply:
But in that situation entire content of that content editable would be selected instead of single paragraph.

Reply:
Note that table cells are also nested editables and triple-click action is expected to select current block only, not entire content editable.

Reply:
You're right, my bad.

Reply:
Can we have another test then that checks only the closest non-attribute is selected and not the entire nested editable?

Reply:
ok